### PR TITLE
langchain-google: remove possible trailing newline from content text in response candidates

### PR DESCRIPTION
**Pull request by aaronbriel**
## PR Description
Renamed `_strip_erroneous_spaces` to `_strip_erroneous_characters` and added trailing newline removal, added unit tests for function.

## Relevant issues

[24648](https://github.com/langchain-ai/langchain/discussions/24648)

## Type

🐛 Bug Fix
🧹 Refactoring
✅ Test

## Testing(optional)

make test

77 passed, 6 warnings in 0.82s

### CODE REVIEW

**Comment by lkuligin** [comment]
PALM is going to be deprecated, do we really need this one?

**Comment by lkuligin** [comment]
actually, if your original issue was with gemini-1.5 model, I'm not sure this PR will fix it.

**Comment by aaronbriel** [comment]
@lkuligin Good catch. I've reverted the earlier code and instead added trailing newline remove to content text of response candidates.

Before change:
```
>>> from langchain_google_genai import ChatGoogleGenerativeAI
>>> import os
>>> llm = ChatGoogleGenerativeAI(google_api_key=os.getenv('GOOGLE_API_KEY'), model='gemini-1.5-pro-002')
>>> prompt = "you are a football expert"
>>> text = "what is the name of the football team in Minnesota?"
>>> messages = [('system', prompt), ('human', text)]
>>> ai_msg = llm.invoke(messages)
>>> ai_msg.content
'The NFL team in Minnesota is the **Minnesota Vikings**.\n'
```

After change:
```
>>> from langchain_google_genai import ChatGoogleGenerativeAI
>>> import os
>>> llm = ChatGoogleGenerativeAI(google_api_key=os.getenv('GOOGLE_API_KEY'), model='gemini-1.5-pro-002')
>>> prompt = "you are a football expert"
>>> text = "what is the name of the football team in Minnesota?"
>>> messages = [('system', prompt), ('human', text)]
>>> ai_msg = llm.invoke(messages)
>>> ai_msg.content
'The NFL team in Minnesota is the **Minnesota Vikings**.'
```

**Comment by aaronbriel** [comment]
@lkuligin are you able to approve this PR or is there someone else to tag?

**Comment by adamJLev** [comment]
waiting for this fix too, thanks for the PR 👌🏼

File: libs/genai/langchain_google_genai/chat_models.py [hunk 1]
```diff
@@ -419,6 +419,8 @@ def _parse_response_candidate(
     for part in response_candidate.content.parts:
         try:
             text: Optional[str] = part.text
+            # Remove erroneous newline character if present
+            text = text.rstrip("\n")
```
Comment (lkuligin) [inline libs/genai/langchain_google_genai/chat_models.py:423]: if text can be None, we should check for this before applying `rstrip`

**Review by lkuligin — CHANGES_REQUESTED** [review]

File: libs/genai/langchain_google_genai/chat_models.py (hunk 1 above)
Comment (aaronbriel) [reply libs/genai/langchain_google_genai/chat_models.py]: I've added a conditional for this.

**Review by lkuligin — APPROVED** [review]

**Pull request merged**